discovery-client/src/mdns-browser/index.ts

```diff
@@ -19,6 +19,18 @@ export type { MdnsBrowser, MdnsBrowserOptions, MdnsBrowserService }
 const IFACE_POLL_INTERVAL_MS = 5000
 const QUERY_INTERVAL_MS = [4000, 8000, 16000, 32000, 64000, 128000]
 
+const robotModelFromTxt = (records: string[] | undefined): string | null => {
+  return records
+    ? records.reduce((cached: string | null, elem: string): string | null => {
+        return (
+          cached ??
+          elem.match(/robotModel=(?<robotModel>.*)/)?.groups?.robotModel ??
+          null
+        )
+      }, null)
+    : null
+}
+
```

Comment:
I've been trying to move away from reduce in my own code, and it's been pretty rewarding in terms of readability. Here, I would recommend a `map` and a `find`:

```suggestion
const RE_ROBOT_MODEL = /robotModel=(?<robotModel>.*)/

const robotModelFromTxt = (records: string[] | undefined): string | null => {
  records = records ?? []

  const model = records
    .map((e: string) => e.match(RE_ROBOT_MODEL)?.groups?.robotModel)
    .find(maybeModel => maybeModel != null)

  return model ?? null
}
```

Reply:
oh yeah that looks very nice